znai-core/src/main/java/com/twosigma/znai/extensions/file/FileIncludePlugin.java

```diff
@@ -50,6 +52,7 @@ public PluginResult process(ComponentsRegistry componentsRegistry,
                                 ParserHandler parserHandler,
                                 Path markupPath,
                                 PluginParams pluginParams) {
+        codeReferences = new CodeReferences(componentsRegistry, pluginParams);
```

Comment:
Don't set a member variable from process().
You implicitly put an ordering of calling method.
process() has to be called before auxiliaryFiles().

Reply:
`process` is by design called before. A lot of plugins depend on this. I am not fully sure I follow what you suggest here.

Reply:
I didn't know that process() has to be called before auxiliaryFiles() by design. My point is that if possible, you force the ordering in your code.